linkerd/app/core/src/errors.rs

```diff
@@ -218,11 +234,13 @@ impl<RspB: Default + hyper::body::HttpBody> respond::Respond<http::Response<RspB
                 }
 
                 let status = http_status(&*error);
+                const SERVER_HEADER: &'static str = concat!("linkerd-proxy/", env!("CARGO_PKG_VERSION"));
```

Comment:
i don't think it's really correct to use the cargo package version as the proxy's version. this is going to be the version of the `linkerd-app-core` crate, which isn't the actual proxy version: https://github.com/aryan9600/linkerd2-proxy/blob/6438d0d19dafc22d6645d198b566085c7c5d0e73/linkerd/app/core/Cargo.toml#L3

instead, we should probably use the `GIT_VERSION` env var, which is the current proxy version git tag, set by the build script: https://github.com/linkerd/linkerd2-proxy/blob/122344732ec1ca60f81cc89ffe31e08a146cd8c9/linkerd/app/core/build.rs#L24-L27

```suggestion
                const SERVER_HEADER: &'static str = concat!("linkerd-proxy/", env!("GIT_VERSION"));
```

Reply:
the linting checks on CI [are failing](https://github.com/linkerd/linkerd2-proxy/pull/1033/checks?check_run_id=2766075149#step:5:363) because the `clippy` linter doesn't like the explicit `'static` lifetime here (constants always have the `'static` lifetime). to get CI passing, we should probably remove it:
```suggestion
                const SERVER_HEADER: &str = concat!("linkerd-proxy/", env!("CARGO_PKG_VERSION"));
```

Reply:
oops, forgot to consult clippy before pushing 😅